src/components/Form/FormArea/index.tsx

```diff
@@ -0,0 +1,39 @@
+import { ReactNode } from 'react'
+import Styled from './index.style'
+
+interface FormAreaProps {
+  children: ReactNode
+  label: string
+  errorMessage?: string
+  currentLength?: number
+  maxLength?: number
+}
+
+const FormArea = ({
+  label,
+  children,
+  errorMessage,
+  maxLength,
+  currentLength = 0,
+}: FormAreaProps) => {
+  return (
+    <Styled.Container>
+      {maxLength ? (
+        <Styled.LabelContainer>
+          <Styled.Label>{label}</Styled.Label>
+          <Styled.Label>
+            {currentLength} / {maxLength}
+          </Styled.Label>
+        </Styled.LabelContainer>
+      ) : (
+        <Styled.Label>{label}</Styled.Label>
+      )}
+      {children}
+      <Styled.ErrorBox>
+        <Styled.ErrorMessage>{errorMessage}</Styled.ErrorMessage>
+      </Styled.ErrorBox>
+    </Styled.Container>
+  )
+}
```

Comment:
```suggestion
interface FormAreaProps {
  children: ReactNode
  label: string
  errorMessage?: string
  currentLength?: number
  maxLength?: number
  labelRight?: React.ReactNode
}

const FormArea = ({
  label,
  children,
  errorMessage,
  maxLength,
  currentLength = 0,
  labelRight,
}: FormAreaProps) => {
  return (
    <Styled.Container>
      <Styled.LabelContainer>
        <Styled.Label>{label}</Styled.Label>
        {labelRight && labelRight}
      </Styled.LabelContainer>
      {children}
      <Styled.ErrorBox>
        <Styled.ErrorMessage>{errorMessage}</Styled.ErrorMessage>
      </Styled.ErrorBox>
    </Styled.Container>
  )
}
```
input을 감싸는 FormArea 컴포넌트를 만드신거 같습니다. maxLength라는 특정 props값을 조건으로 렌더링하는데, 특정값을 조건으로 렌더링하게되면 컴포넌트가 유연하기 힘들다고 생각합니다. 만약 기획과 디자인 요구상 해당 위치에 다른 컴포넌트가 들어오게된다면 또다시 props를 뚫고 조건을 걸어야겠죠. 
해당위치에 리액트 컴포넌트를 받을 수 있도록 열어놓고 외부에서 필요한 컴포넌트를 주입받는것이 좋을 것 같습니다.

Reply:
> input을 감싸는 FormArea 컴포넌트를 만드신거 같습니다. maxLength라는 특정 props값을 조건으로 렌더링하는데, 특정값을 조건으로 렌더링하게되면 컴포넌트가 유연하기 힘들다고 생각합니다. 만약 기획과 디자인 요구상 해당 위치에 다른 컴포넌트가 들어오게된다면 또다시 props를 뚫고 조건을 걸어야겠죠.
해당위치에 리액트 컴포넌트를 받을 수 있도록 열어놓고 외부에서 필요한 컴포넌트를 주입받는것이 좋을 것 같습니다.

변경에 너무 막혀있는 컴포넌트를 만든 것 같네요.. 말씀하신 대로 `ReactNode`를 받아서 해당 위치에 넣을 수 있게 변경하는 게 맞는 방향 같습니다!